web/src/user_group_edit.ts

```diff
@@ -467,19 +524,19 @@ function get_membership_status_context(group) {
     };
 }
 
-function update_membership_status_text(group) {
+function update_membership_status_text(group: UserGroup): void {
     const args = get_membership_status_context(group);
     const rendered_membership_status = render_user_group_membership_status(args);
     const $edit_container = get_edit_container(group);
     $edit_container.find(".membership-status").html(rendered_membership_status);
 }
 
-export function show_settings_for(group) {
+export function show_settings_for(group: UserGroup): void {
     const html = render_user_group_settings({
         group,
         // We get timestamp in seconds from the API but timerender needs milliseconds.
         date_created_string: timerender.get_localized_date_or_time_for_format(
-            new Date(group.date_created * 1000),
+            new Date((group.date_created ?? 0) * 1000),
```

Comment:
So structurally, the logic here is assuming that `group.date_created` is only to be displayed if `creator` is defined; basically it's a fact about the API that we have both or neither.

Probably this deserves a clarifying comment that this `0` value will never be used.

Reply:
Added a bonus commit with this comment.